File: app/src/main/java/org/metabrainz/mobile/api/data/search/entity/LifeSpan.java
package org.metabrainz.mobile.api.data.search.entity;

import java.io.Serializable;

public class LifeSpan implements Serializable {
    private String begin;
    private String end;
    private boolean ended;

    @Override
    public String toString() {
        return "LifeSpan{" +
                "begin='" + begin + '\'' +
                ", end='" + end + '\'' +
                ", ended=" + ended +
                '}';
    }

    public String getBegin() {
        return begin;
    }

    public void setBegin(String begin) {
        this.begin = begin;
    }

    public String getEnd() {
        return end;
    }

    public void setEnd(String end) {
        this.end = end;
    }

    public boolean isEnded() {
        return ended;
    }

    public void setEnded(boolean ended) {
        this.ended = ended;
    }

    public String getTimePeriod() {
        StringBuilder builder = new StringBuilder();
        if (begin != null && !begin.isEmpty()) builder.append(begin);
        if (begin != null && !begin.isEmpty() && end != null && !end.isEmpty()) builder.append("-");
        if (end != null && !end.isEmpty()) builder.append(end);
        return builder.toString();
    }
}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/LifeSpan.kt
package org.metabrainz.mobile.data.sources.api.entities

import java.io.Serializable

class LifeSpan : Serializable {
    var begin: String? = null
    var end: String? = null
    var isEnded = false
    override fun toString(): String {
        return "LifeSpan{" +
                "begin='" + begin + '\'' +
                ", end='" + end + '\'' +
                ", ended=" + isEnded +
                '}'
    }

    val timePeriod: String
        get() {
            val builder = StringBuilder()
            if (begin != null && !begin!!.isEmpty()) builder.append(begin)
            if (begin != null && !begin!!.isEmpty() && end != null && !end!!.isEmpty()) builder.append("-")
            if (end != null && !end!!.isEmpty()) builder.append(end)
            return builder.toString()
        }
}